resources/netdata_install.rb

```diff
@@ -30,58 +35,137 @@
 
-  execute 'install' do
-    cwd new_resource.git_source_directory
-    command <<-EOF
-      ./netdata-installer.sh #{enable_autoupdate} #{custom_install_path} \
-      --zlib-is-really-here --dont-wait --dont-start-it
-    EOF
-    notifies :install, 'package[compile_packages]', :before
-    notifies :restart, 'service[netdata]', :delayed
-    not_if { autoupdate_enabled_on_system? }
-    only_if { update || !netdata_installed? }
-  end
+    git new_resource.git_source_directory do
+      repository new_resource.git_repository
+      reference new_resource.git_revision
+      action :sync
+    end
+
+    execute 'install' do
+      cwd new_resource.git_source_directory
+      command <<-EOF
+        ./netdata-installer.sh #{enable_autoupdate} #{custom_install_path} \
+        --zlib-is-really-here --dont-wait --dont-start-it
+      EOF
+      notifies :install, 'package[compile_packages]', :before
+      notifies :restart, 'service[netdata]', :delayed
+      not_if { autoupdate_enabled_on_system? }
+      only_if { update || !netdata_installed? }
+    end
+
+    service 'netdata' do
+      supports status: true, restart: true, reload: true
+      action [:start, :enable]
+    end
+
+  when 'binary'
+    unless %w(amd64 x86_64).include?(node['kernel']['machine'])
+      raise "Static binary versions of netdata are available only for 64bit Intel/AMD CPUs (x86_64), but yours is: #{node['kernel']['machine']}."
+    end
+
+    binary_package = new_resource.binary_package
+    binary_repository = new_resource.binary_repository
+
+    if binary_package == 'netdata-latest.gz.run'
+      remote_file "#{Chef::Config[:file_cache_path]}/netdata_latest.txt" do
+        source "#{binary_repository}/#{binary_package}"
+        action :create
+      end
+      ruby_block 'set_netdata_binary_package_env' do
+        block do
+          ENV['NETDATA_BINARY_PACKAGE'] = ::File.read("#{Chef::Config[:file_cache_path]}/netdata_latest.txt")
+        end
+      end
+      remote_file "#{Chef::Config[:file_cache_path]}/netdata-package.gz.run" do
+        source lazy { "#{binary_repository}/#{::File.read("#{::Chef::Config[:file_cache_path]}/netdata_latest.txt")}" }
+        action :create
+      end
+      file "#{Chef::Config[:file_cache_path]}/netdata_latest.txt" do
+        action :delete
+      end
+    else
+      ruby_block 'set_netdata_binary_package_env' do
+        block do
+          ENV['NETDATA_BINARY_PACKAGE'] = binary_package
+        end
+      end
+      remote_file "#{Chef::Config[:file_cache_path]}/#{binary_package}" do
+        source "#{binary_repository}/#{binary_package}"
+        action :create
+      end
+    end
+
+    options = new_resource.binary_options
+
+    # check if options include "--accept", otherwise automated installation will fail
+    options.insert(0, '--accept') unless options.include?('--accept')
+
+    # place '--dont-start-it' in the end with '--' to pass to post-installer.sh
+    if options.include?('--dont-start-it')
```

Comment:
In regards to the logic with options array, I think if the engineer is setting different options then what are the default options they'll understand what they are doing and the cookbook doesn't have to try and protect them.

Also, these items could change easily in the future or there could be a use case that these don't work with and causes someone a problem that they can't get out of because of this logic.

Probably best to put it in the README and provide some Chef warning logs. It'll be very difficult to maintain this level of granularity over time.

Reply:
this is the logic of the kickstart-static script. you are right it should be mentioned in README
We have to support this logic if we use binary packages from the git repository.
Other option is to use the kickstart-static scipt in `netdata_install` resource:
`bash <(curl -Ss https://my-netdata.io/kickstart-static64.sh) --dont-wait`
Do you think this is more appropriate?
For my use case I just put the binary package somewhere close to nodes and bootstrap them with Netdata, so the usage of binary packages is more desirable.
Chef warning will be provided.
And if we test this regularly that won't be a problem.

Reply:
Agreed testing this regularly won't be an issue. I'm unfamiliar with the kickstart or binary installs, so it was more of me missing knowledge about it then a problem with the code. 

Makes more sense, an update to the README would help clarify the logic or a comment in the code.

Reply:
true, install_options and post_install_options are now in place with similar logic